addons/website/models/website.py

```diff
@@ -126,10 +127,16 @@ def _onchange_language_ids(self):
         if language_ids and self.default_lang_id not in language_ids:
             self.default_lang_id = language_ids[0]
 
+    @api.depends('menu_ids.parent_id')
     def _compute_menu(self):
-        Menu = self.env['website.menu']
+        # accessing through menu_ids allows to prefetch all menus at once
+        self.menu_ids.child_id
         for website in self:
-            website.menu_id = Menu.search([('parent_id', '=', False), ('website_id', '=', website.id)], order='id', limit=1).id
+            website.menu_id = False
```

Comment:
the default assignation can be done once before the for loop ;)
`self.menu_id = False`

Reply:
This is still a POC, the whole code got replaced, but thanks